pypesto/optimize/optimizer.py

```diff
@@ -936,8 +936,16 @@ def minimize(
         args = {'mode': MODE_FUN}
         if self.hessian_update is None or isinstance(self.hessian_update,
                                                      fides.HybridUpdate):
+            if not problem.objective.has_hess:
+                raise ValueError('Specified hessian update scheme cannot be '
+                                 'used with objectives that do not support '
+                                 'Hessian computation.')
             args['sensi_orders'] = (0, 1, 2)
         else:
+            if not problem.objective.has_grad:
```

Comment:
the gradient question should not require an "else", right?

Reply:
correct